addons/mail/models/discuss/ir_websocket.py

```diff
@@ -79,28 +68,72 @@ def _build_bus_channel_list(self, channels):
         domain = ["|", ("is_member", "=", True), ("id", "in", discuss_channel_ids)]
         all_user_channels = self.env["discuss.channel"].search(domain)
         member_specific_channels = [(c, "members") for c in all_user_channels if c.id not in discuss_channel_ids]
-        channels.extend([*all_user_channels, *member_specific_channels])
+        channels.extend([*all_user_channels, *member_specific_channels, *presence_channels])
         return super()._build_bus_channel_list(channels)
 
+    def _prepare_subscribe_data(self, channels, last):
+        data = super()._prepare_subscribe_data(channels, last)
+        data["channels"].update(self._build_bus_channel_list(channels))
+        return data
+
+    def _dispatch_missed_presences(self, channels):
+        # There is a gap between a subscription client side (which is debounced)
+        # and the actual subcription thus presences can be missed. Send a
+        # notification to avoid missing presences during a subscription.
+        presence_channels = [c for c in channels if isinstance(c, tuple) and c[1] == "presence"]
+        partners = self.env["res.partner"].browse(
+            [p.id for p, _ in presence_channels if isinstance(p, self.pool["res.partner"])]
+        )
+        guest_ids = [g.id for g, _ in presence_channels if isinstance(g, self.pool["mail.guest"])]
+        domain = expression.AND(
+            [
+                [("last_poll", ">", datetime.now() - timedelta(seconds=2))],
+                expression.OR(
+                    [
+                        [("user_id", "in", partners.with_context(active_test=False).sudo().user_ids.ids)],
+                        [("guest_id", "in", guest_ids)]
+                    ]
+                ),
+            ]
+        )
+        # sudo: mail.presence: can access presences linked to presence channels.
+        missed_presences = self.env["mail.presence"].sudo().search(domain)
+        bus_target = (
```

Comment:
`bus_target` could be an empty recordset, shouldn't we handle this case?

Reply:
I am not sure how it could be an empty record set, public users which are not guests?